### PR TITLE
v7: npm view

### PR DESCRIPTION
**Pull request by claudiahdz**
npm view refactor and unit tests

### CODE REVIEW

File: test/lib/view.js [hunk 1]
```diff
@@ -0,0 +1,492 @@
+const t = require('tap')
+const mr = require('npm-registry-mock')
```
Comment (ruyadorno) [inline test/lib/view.js:2]: just a tip, I have been finding it way simpler to just mock the lib that reaches for the registry in the case of unit tests (in this case `pacote`) then setting up a complete registry mock

File: tap-snapshots/test-lib-view.js-TAP.test.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+`
+
+exports[`test/lib/view.js TAP should log package info package with no repo or homepage > must match snapshot 1`] = `
+
+
+[4m[1m[32mblue[39m@[32mundefined[39m[22m[24m | [1m[31mProprietary[39m[22m | deps: [32mnone[39m | versions: [33m2[39m
+
+dist
+.tarball:[36mundefined[39m
+.shasum:[33mundefined[39m
```
Comment (ruyadorno) [inline tap-snapshots/test-lib-view.js-TAP.test.js:280]: there are quite a few `undefined` all over the snapshots, are these expected due to the nature of the mocked data?

File: tap-snapshots/test-lib-view.js-TAP.test.js [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+[33m22[39m: [object Object]
+[33m23[39m: [object Object]
+[33m2[39m: [object Object]
+[33m3[39m: [object Object]
+[33m4[39m: [object Object]
+[33m5[39m: [object Object]
+[33m6[39m: [object Object]
+[33m7[39m: [object Object]
+[33m8[39m: [object Object]
+[33m9[39m: [object Object]
```
Comment (ruyadorno) [inline tap-snapshots/test-lib-view.js-TAP.test.js:251]: were these supposed to be the name?

**Comment by isaacs** [comment]
Landed on 87d27d389065609e94ee218ab001972fc0041fe9.  Thanks!

**Pull request closed without merging**